### PR TITLE
feat: [PL-27997]: Add error component in wizard final step

### PR DESCRIPTION
**Pull request by upadhyay-an**
### Summary
Instead of showing the create/update error in a toaster now it comes as part of Error component on last screen itself.
Refactored code so that any step can be configured to save data and show any resulting error.

Fixed other minor bugs as well. (Wizard max width & query index reversal)

#### Screenshots
Screenshot at Sep 19 02-52-13

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 10 Issues 
* Code issue 10 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11289)

**Comment by upadhyay-an** [comment]
retrigger titlecheck

**Review by pbarapatre10 — APPROVED** [review]
LGTM

**Comment by meenakshi-harness** [comment]
@upadhyay-an  Most of the places we display error at the top of the modal. Is it intentional to show at the bottom here?

File: src/modules/33-auth-settings/components/CreateUpdateLdapWizard/utils.tsx [hunk 1]
```diff
@@ -101,3 +101,10 @@ export const QueryFormTitle: React.FC<{ title: string }> = ({ title }) => {
     </>
   )
 }
+
+export const getErrorMessageFromException = (e: any, fallbackMessage?: string): ResponseMessage[] => {
+  const hasResponseMessages = e.data?.responseMessages && e.data?.responseMessages.length > 0
+  return hasResponseMessages
+    ? e.data?.responseMessages
+    : [{ level: 'ERROR', message: e.data?.message || e.message || fallbackMessage }]
+}
```
Comment (meenakshi-harness) [inline src/modules/33-auth-settings/components/CreateUpdateLdapWizard/utils.tsx:110]: Can we check if we already don't have similar util present?

**Comment by mayankVermaHarness** [comment]
@upadhyay-an Is this in-line with previous designs? Usually errors are shown on top of modals—instead of bottom.

**Comment by upadhyay-an** [comment]
@meenakshi-harness @mayankVermaHarness  This wizard step has bunch of CTAs which (testing query, committing query and saving all data) which could result in UI Error component being rendered if the backend call fails.

At each of the places we show Error component near the corresponding CTA button, this is as per the discussion with product team.

File: src/modules/33-auth-settings/components/CreateUpdateLdapWizard/utils.tsx (hunk 1 above)
Comment (meenakshi-harness) [reply src/modules/33-auth-settings/components/CreateUpdateLdapWizard/utils.tsx:110]: Can you check `getErrorInfoFromErrorObject`

**Review by meenakshi-harness — APPROVED** [review]

**Pull request merged**